src/provider/JSONRPC.ts

```diff
@@ -40,7 +41,7 @@ export enum JSONRPCMethod {
   eth_getFilterLogs = "eth_getFilterLogs"
 }
 
-export interface JSONRPCRequest<T = any[]> {
+export interface JSONRPCRequest<T = any> {
```

Comment:
Changing here in order to avoid a typescript error on the dapps side. On wallet_watchAsset ethereum request, the field params is not an array (i'm not sure why it has changed for this one), so JSONRPCRequest now should expect both arrays and objects.

[image](https://user-images.githubusercontent.com/26755665/154082241-5d72dc25-ce35-48c3-8a9e-60d7d32be126.png)

Ref: [EIP-747](https://eips.ethereum.org/EIPS/eip-747),